File: Data Structures & Algorithms/count-subsequences/submission-0.cpp
class Solution {
public:
    int numDistinct(string s, string t) {
        vector<vector<int>> dp(t.length()+1,vector<int>(s.length()+1,0));
       
        for(int j=0;j<=s.length();j++){
            dp[t.length()][j] = 1;
        }
        for(int i=t.length()-1;i>=0;i--){
            for(int j=s.length()-1;j>=0;j--){
                dp[i][j] = dp[i][j+1];
                if(t[i] == s[j]){
                    dp[i][j] += dp[i+1][j+1];
                }
            }
        }
        return dp[0][0];
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-0.cs
public class Solution {
    public int LeastInterval(char[] tasks, int n) {
        PriorityQueue<int,int> maxHeap = new PriorityQueue<int,int>();
        int[] count = new int[26];
        foreach(char task in tasks){
            count[task -'A']++;
        }
        for(int i =0;i<26;i++){
            if(count[i] > 0){
                maxHeap.Enqueue(count[i],-count[i]);
            }
        }
        Queue<int[]> queue = new Queue<int[]>();

        int time = 0;
        while(maxHeap.Count > 0 || queue.Count > 0){
            if(queue.Count > 0 && time >= queue.Peek()[1]){
                int[] temp = queue.Dequeue();
                maxHeap.Enqueue(temp[0],-temp[0]);
            }
            if(maxHeap.Count>0){
                int cnt = maxHeap.Dequeue() -1;
                if(cnt > 0){
                    queue.Enqueue(new int[]{cnt,time+n+1});
                }
            }
            time++;
        }
        return time;
    }
}
